Lib/test/test_largefile.py

```diff
@@ -151,9 +151,22 @@ def test_seekable(self):
                 self.assertTrue(f.seekable())
 
 
+def skip_no_disk_space(path, required):
+    def decorator(fun):
+        def wrapper(*args, **kwargs):
+            if shutil.disk_usage(os.path.realpath(path)).free < required:
+                hsize = int(required / 1024 / 1024)
+                raise unittest.SkipTest(
+                    f"required {hsize} MiB of free disk space")
+            return fun(*args, **kwargs)
+        return wrapper
+    return decorator
+
+
 class TestCopyfile(LargeFileTest, unittest.TestCase):
     open = staticmethod(io.open)
 
+    @skip_no_disk_space(TESTFN, size * 3)
```

Comment:
Why is it *3? I count only two files: TESTFN and TESTFN2. Maybe add a comment explaining why it's *3?

Reply:
It's just for extra safety. The required **exact** disk space would be `(file size * 2)` (aka 4.7G), but just in case something else writes on disk (especially on builtbots) I opted for `(file size * 3)` (around 7.2 GB).

Reply:
I added a comment:
```
    # Exact required disk space would be (size * 2), but let's give it a
    # bit more tolerance.
```

Reply:
If size*2 is enough, I suggest to use size*2.5 for safety rather than size*3 which is quite huge.

Reply:
Makes sense. Done.